client/settings/general-settings-section/payment-methods-list.js

```diff
@@ -65,45 +70,67 @@ const GeneralSettingsSection = () => {
 		if ( hasBeenChecked ) {
 			setEnabledPaymentMethods( [ ...enabledPaymentMethods, method ] );
 		} else {
-			setEnabledPaymentMethods(
-				enabledPaymentMethods.filter( ( m ) => m !== method )
-			);
+			setIsConfirmationModalOpen( true );
+			setModalOpenForMethod( method );
 		}
 	};
 
+	const handleRemoveMethod = ( method ) => {
+		setIsConfirmationModalOpen( false );
+		setModalOpenForMethod( null );
+		setEnabledPaymentMethods(
+			enabledPaymentMethods.filter( ( m ) => m !== method )
+		);
+	};
+
 	return (
-		<List>
-			{ availablePaymentMethods.map( ( method ) => {
-				const { Icon, label, description } = PaymentMethodsMap[
-					method
-				];
-
-				return (
-					<ListElement key={ method }>
-						{ isUpeEnabled && (
-							<CheckboxControl
-								label={
-									<VisuallyHidden>{ label }</VisuallyHidden>
-								}
-								onChange={ makeCheckboxChangeHandler( method ) }
-								checked={ enabledPaymentMethods.includes(
-									method
-								) }
-							/>
-						) }
-						<PaymentMethodWrapper>
-							<PaymentMethodDescription
-								id={ isUpeEnabled ? method : null }
-								Icon={ Icon }
-								description={ description }
-								label={ label }
-							/>
-							{ isUpeEnabled && <StyledFees id={ method } /> }
-						</PaymentMethodWrapper>
-					</ListElement>
-				);
-			} ) }
-		</List>
+		<>
+			<List>
+				{ availablePaymentMethods.map( ( method ) => {
+					const { Icon, label, description } = PaymentMethodsMap[
+						method
+					];
+
+					return (
+						<ListElement key={ method }>
+							{ isUpeEnabled && (
+								<CheckboxControl
+									label={
+										<VisuallyHidden>
+											{ label }
+										</VisuallyHidden>
+									}
+									onChange={ makeCheckboxChangeHandler(
+										method
+									) }
+									checked={ enabledPaymentMethods.includes(
+										method
+									) }
+								/>
+							) }
+							<PaymentMethodWrapper>
+								<PaymentMethodDescription
+									id={ isUpeEnabled ? method : null }
+									Icon={ Icon }
+									description={ description }
+									label={ label }
+								/>
+								{ isUpeEnabled && <StyledFees id={ method } /> }
+							</PaymentMethodWrapper>
+						</ListElement>
+					);
+				} ) }
+				{ isConfirmationModalOpen && (
+					<RemoveMethodConfirmationModal
+						method={ modalOpenForMethod }
+						onClose={ () => setIsConfirmationModalOpen( false ) }
+						handleRemove={ () =>
+							handleRemoveMethod( modalOpenForMethod )
+						}
+					/>
+				) }
+			</List>
+		</>
```

Comment:
Looks like `<List></List>` is already wrapping the component. We can probably remove `<></>`.